p32/src/bank.rs

```diff
@@ -0,0 +1,249 @@
+#[derive(Clone, Debug)]
+/// Data type to hold users
+pub struct User {
+    name: String,
+    credit_line: u64,
+    balance: i64,
+}
+
+#[derive(Debug)]
+/// Data type to hold banks
+pub struct Bank {
+    users: Vec<User>,
+    pub name: String,
+    credit_interest: u64,
+    debit_interest: u64,
+}
+
+pub enum Error {
+    UserNotFound { username: String },
+    BalanceOverflow { username: String },
+    BalanceUnderflow { username: String },
+    BadAmount { amount: u64 },
+    CreditLimitReached,
+    LiabilitiesOverflow,
+    AssetsOverflow,
+    InterestOverflow { username: String },
+    BadInterest { interest: u64 },
+}
+
+/// Implementation of methods for bank
+impl Bank {
+    /// Compute total liabilities and assets and return as a tuple
+    pub fn calc_balance(&self) -> Result<(u64, u64), Error> {
+        let mut liabilities: u64 = 0;
+        let mut assets: u64 = 0;
+        for user in self.users.iter() {
+            if user.balance > 0 {
+                liabilities = match liabilities.checked_add(user.balance.unsigned_abs()) {
+                    Some(b) => b,
+                    None => return Err(Error::LiabilitiesOverflow),
+                }
+            } else {
+                assets = match assets.checked_add(user.balance.unsigned_abs()) {
+                    Some(b) => b,
+                    None => return Err(Error::AssetsOverflow),
+                }
+            }
+        }
+        Ok((liabilities, assets))
+    }
+
+    /// Transfer fund from debit-user to credit-user
+    pub fn transfer_funds(
+        &mut self,
+        username_debit: String,
+        username_credit: String,
+        amount: u64,
+    ) -> Result<bool, Error> {
```

Comment:
This method can return `Result<(), Error>` since on success it always returns `Ok(true)`. Plus you can remove this method, since you already have a better alternative implementation below.